docs/docs/typescript/strict-mode.md

```diff
@@ -90,26 +75,66 @@ One of the challenges in the GraphQL-Prisma world is the difference in the way t
 - but For Prisma, `null` is a value, and `undefined` means "do nothing"
 
 This is covered in detail in [Prisma's docs](https://www.prisma.io/docs/concepts/components/prisma-client/null-and-undefined), which we strongly recommend reading.
-But the gist of it is that, for Prisma's create and update operations, you have to make sure `null`s are converted to `undefined` from your GraphQL mutation inputs.
-One way to do this is to use the [dnull](https://www.npmjs.com/package/dnull) package:
+But the gist of it is that, for Prisma's create and update operations, you have to make sure `null`s are converted to `undefined` from your GraphQL mutation inputs - unless the field is nullable in your Prisma schema.
 
-```
-yarn workspace api add dnull
-```
+One way to do this is to use the `removeNulls` utility function from `@redwoodjs/api`
 
 ```ts title=api/src/services/users.ts
 // highlight-next-line
-import { dnull } from "dnull"
+import { removeNulls } from "@redwoodjs/api"
 
 export const updateUser: MutationResolvers["updateUser"] = ({ id, input }) => {
   return db.user.update({
     // highlight-next-line
-    data: dnull(input),
+    data: removeNulls(input),
     where: { id },
   })
 }
 ```
 
+### Relation resolvers in services
```

Comment:
I don't like this recommendation here much..... but what else do we do?

The problem is that `findUnique` _can_ return null - but inside the relation resolver it can't, because if the Book wasn't found, it wouldn't have called the relation resolver in the first place!